src/main/java/org/minions/devfund/noemiguzman/battleship/Ship.java

```diff
@@ -0,0 +1,219 @@
+                    try {
+
+                        if (!ocean.getShipArray()[i][j].getShipType().equals("empty")) {
+                            return false;
+                        }
+                    } catch (Exception e) {
+                        continue;
+                    }
+                }
+            }
+        }
+
+        return true;
+    }
+
+    /**
+     * Puts the ship in the ocean.
+     * This involves giving values to the bowRow, bowColumn, and horizontal instance variables in the ship.
+     * @param row number
+     * @param column number
+     * @param horizontal true
+     * @param ocean object
+     */
+    public void placeShipAt(int row, int column, boolean horizontal, final Ocean ocean) {
+        bowRow = row;
+        bowColumn = column;
+        this.horizontal = horizontal;
+
+        /*
+         * Set references to the ship in the ships array in the Ocean object
+         */
+        if (horizontal) {
+            for (int j = column; j < column + getLength(); j++) {
+                ocean.getShipArray()[row][j] = this;
+            }
+        } else {
+            for (int i = row; i < row + getLength(); i++) {
+                ocean.getShipArray()[i][column] = this;
+            }
+        }
+    }
+
+
+    /**
+     * Return true if the part of the ship was hit, false otherwise.
+     * @param row number
+     * @param column number
+     * @return true
+     */
+    public boolean wasShootAt(int row, int column) {
+        if (horizontal) {
+            return hit[column - this.bowColumn];
+        } else {
+            return hit[row - this.bowRow];
+        }
+    }
+
+    /**
+     * If a part of the ship occupies the given row and column, and the ship hasnot been sunk,
+     * mark that part of the ship as hit (in the hit array, 0 indicates the bow) and return true,
+     * otherwise return false.
+     * @param row number
+     * @param column number
+     * @return true
+     */
+    public boolean shootAt(int row, int column) {
+        if (!isSunk()) {
+            if (horizontal) {
+                if (row == this.bowRow && column < this.bowColumn + length) {
+                    hit[column - this.bowColumn] = true;
+                    return true;
+                }
+            } else {
+                if (column == this.bowColumn && row < this.bowRow + length) {
+                    hit[row - this.bowRow] = true;
+                    return true;
+                }
+            }
+        }
+        return false;
+    }
+
+
+    /**
+     * Return true if every part of the ship has been hit, false otherwise.
+     * @return true
+     */
+    public boolean isSunk() {
+        for (boolean b : hit) {
+            if (!b) {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    /*
+     * Return X if the ship has been sunk,  S if it has not been sunk.
+     * This method can be used to print out locations in the ocean that have been shot at;
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') This block of commented-out lines of code should be removed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ACommentedOutCodeLine)